src/providers/mbed_provider/utils.rs

```diff
@@ -81,13 +85,11 @@ pub fn convert_key_bits(key_size: u32) -> psa_key_bits_t {
 /// # Panics
 ///
 /// Only `KeyType::RsaKeypair` and `KeyType::RsaPublicKey` are supported. Panics otherwise.
-pub fn convert_key_type(key_type: KeyType) -> psa_key_type_t {
+pub fn convert_key_type(key_type: KeyType) -> Result<psa_key_type_t> {
     match key_type {
-        KeyType::RsaKeypair => PSA_KEY_TYPE_RSA_KEYPAIR,
-        KeyType::RsaPublicKey => PSA_KEY_TYPE_RSA_PUBLIC_KEY,
-        _ => {
-            unimplemented!();
-        }
+        KeyType::RsaKeypair => Ok(PSA_KEY_TYPE_RSA_KEYPAIR),
+        KeyType::RsaPublicKey => Ok(PSA_KEY_TYPE_RSA_PUBLIC_KEY),
+        _ => Err(ResponseStatus::PsaErrorInvalidArgument),
```

Comment:
Wouldn't this be `UnsupportedParameters` as well?

Reply:
True, will change.